data/jet/config/config.txt
####################
#      Config      #
####################

# Use "#" for comments.
# Do *not* remove variables or change their names,
# otherwise the generation code will not work
# (and you will have to fix/redownload this file!)


# 1) Process configuration.

	# 1a) Chosen process

	# Give a comma-separated list (spaces allowed). 
	# See the samples subdirectory names to see which 
	# processes are available. The names must match the
	# subdirectory names, and there must be only one
	# template per subdirectory.

	#processes = Top_Wqq, QCD
        processes = Top_Wqq

	# 1b) pT boundaries

	# Give pT values to use as bin edges for 
	# the different samples. Each pT bin will
	# correspond with one condor job.
	# For infinity (valid as an upper bound), add
	# 'inf' to the end of this list.

	pT_cutoffs = 550,560,570,580,590,600,610,620,630,640,650
	#pT_cutoffs = 550,560

	# 1c) Number of events
	
	# Give the number of events to generate for
	# each pT window.

	Nevents = 100000

	# 1d) Hadronization

	# Determine whether or not to perform hadronization.
	# 1 = on, 0 = off.

	hadronization = 1

	# 1e) Multi-Parton Interactions

	# Determine whether or not to simulate MPI.
	# 1 = on, 0 = off

	mpi = 0

	# 1f) Random Seed

	# Seed for the Pythia8 RNG. Use this to generate
	# unique samples for the same configuration.
	# Must be integer greater than 0.

	rng = 1

# 2) Jet clustering

	#2a) Power
	
	# Give a "power" for the generalized kT algo.
	# -1 = anti-kt
	#  0 = Cambrige/Aachen
	#  1 = kt

	power = -1

	#2b) Radius

	# Give the jet radius in eta-phi space.
	
	radius = 0.8

	#2c) Minimum pT

	# Give the minimum jet pT.

	pT_min = 15.

	#2d) Maxinimum eta

	# Give the maximum (magnitude) of eta.
	# For reference, eta_max = 4.9 corresponds
	# with the maximum possible value for ALTAS,
	# based on the LAr calorimeter geometry.
	
	eta_max = 2.

	#2e) Number of constituents

	# Give the maximum number of jet 
	# constituents to save to our
	# list of constituents. This is the
	# the data that the Lorentz Group Network
	# will actually use. For reference,
	# n_constituents = 200 for the top-tagging
	# reference dataset located at
	# https://zenodo.org/record/2603256 .

	n_constituents = 200

# 3) Truth-level & Final-state Particles

        #3a) Number of truth-level particles

	# Give the maximum number of truth-level
        # particles to save for each event.

        n_truth = 3

        #3b) Number of final-state particles

        # Give the maximum number of final-state
        # particles to save for each event.
        # (This probably doesn't need to be changed)

        n_final = 4000



data/jet/config/jet_parameters/jet_parameters.txt
#Jet finder settings. All lines containing "#" will be treated as comments.
power = -1
R = 0.8
pT_min = 15.
eta_max = 2.
n_constituents = 200


data/jet/generation/utils/condor/config.py

# config.py

# Helper functions for setting up configuration for event generation.

def GetJetParameters(path_to_file = 'jet_parameters.txt'):
    jet_vars = {} # entries are of form [value, found?, [keywords], datatype]
    jet_vars['radius'] = [0.8, False,['r','R'], 'float']
    jet_vars['power'] = [-1, False,['power','Power'], 'int']
    jet_vars['min_pt'] = [15., False, ['pT','pt','PT','Pt'], 'float']
    jet_vars['max_eta'] = [2., False, ['eta','Eta'], 'float']
    jet_vars['nconst'] = [200, False, ['const','Const'], 'int']
    
    with open(path_to_file,'r') as f:
        for line in f.readlines():
            if (not '=' in line): continue
            name = line.split('=')[0]
            value = line.split('=')[1]
            for key, val in jet_vars.items():
                for keyword in val[2]:
                    if keyword in line:
                        val[1] = True
                        if(val[3] == 'int'): val[0] = int(value)
                        else: val[0] = float(value)
    
    for key, val in jet_vars.items():
        if(val[1] == False): print('Warning: Did not find a value for ', key, ' in ', path_to_file)
    return jet_vars

def GetNEvents(path_to_file = 'config.txt'):
    with open(path_to_file,'r') as f:
        for line in f.readlines():
            if (not '=' in line): continue
            name = line.split('=')[0].strip()
            value = line.split('=')[1].replace('\n','').strip()
            if ('N' in name):
                return int(value)
    print('Warning: Did not find number of events in ', path_to_file)
    return 1

def GetNTruth(path_to_file = 'config.txt'):
    with open(path_to_file,'r') as f:
        for line in f.readlines():
            if (not '=' in line): continue
            name = line.split('=')[0].strip()
            value = line.split('=')[1].replace('\n','').strip()
            if ('n_truth' in name):
                return int(value)
    print('Warning: Did not find number of truth-level particles in ', path_to_file)
    return 10 # default

def GetNFinal(path_to_file = 'config.txt'):
    with open(path_to_file,'r') as f:
        for line in f.readlines():
            if (not '=' in line): continue
            name = line.split('=')[0].strip()
            value = line.split('=')[1].replace('\n','').strip()
            if ('n_final' in name):
                return int(value)
    print('Warning: Did not find number of final-state particles in ', path_to_file)
    return 4000 # default

def GetHadronization(path_to_file = 'config.txt'):
    with open(path_to_file,'r') as f:
        for line in f.readlines():
            if (not '=' in line): continue
            name = line.split('=')[0].strip()
            value = line.split('=')[1].replace('\n','').strip()
            if ('hadronization' in name):
                return (int(value) > 0)
    print('Warning: Did not find hadronization config in ', path_to_file)
    return True

def GetMPI(path_to_file = 'config.txt'):
    with open(path_to_file,'r') as f:
        for line in f.readlines():
            if (not '=' in line): continue
            name = line.split('=')[0].strip()
            value = line.split('=')[1].replace('\n','').strip()
            if ('mpi' in name):
                return (int(value) > 0)
    print('Warning: Did not find MPI config in ', path_to_file)
    return False

def GetRngSeed(path_to_file = 'config.txt'):
    with open(path_to_file,'r') as f:
        for line in f.readlines():
            if (not '=' in line): continue
            name = line.split('=')[0].strip()
            value = line.split('=')[1].replace('\n','').strip()
            if ('rng' in name):
                return int(value)
    print('Warning: Did not find RNG seed in ', path_to_file)
    return 1
